datadayessentials/modelling/models/tests/test_model_factory.py

```diff
@@ -0,0 +1,21 @@
+import unittest
+from ..model_factory import ModelFactory
+
+
+class TestGetLatestRunIdFromExperiment(unittest.TestCase):
+    def test_get_latest_run_id_from_experiment_single_run(self):
+        latest_run_id = ModelFactory.get_latest_run_id_from_experiment(
+            "Ratesetter_auto_ml_run"
+        )
+        print(latest_run_id, "********")
+        assert isinstance(latest_run_id, str)
+
+    def test_get_latest_run_id_from_false_experiment(self):
+        with self.assertRaises(Exception):
+            latest_run_id = ModelFactory.get_latest_run_id_from_experiment(
+                "IM NOT AN EXPERIMENT"
+            )
+
+    def test_get_latest_run_id_from_none(self):
+        with self.assertRaises(Exception):
```

Comment:
raising an exception is a bit of a catch all.  Might be better to have something that raises a specific error if an unrecognised experiment is passed

Reply:
you would need to find exactly what type of error is raised for this situation / raise the error yourself if an incorrect experiment name is passed (maybe with a specific error message), that way your test highlights that the issue was that the experiment name was wrong.  Otherwise lets say for example you passed INCORRECT_EXPERIMENT_NAME.  The issue here might be for examp[le that Azure doesnt allow underscores in experiment names.  Your test will work because an exception was raised, but not for the reason you thought